### PR TITLE
Removed error in newer C# standards.

### PR DESCRIPTION
**Pull request by master117**
Apparently:
?? must return a value not throw.
is does not contain a new var creation.
out does not not allow for creation of a new variable.

Project has errors when loading in Visual Studio 2015/2017, project file filetype is outdated. I could create a new project file if wanted.

### CODE REVIEW

**Comment by DamienDennehy** [comment]
None of these changes are required. 
The project builds fine in Visual Studio 2017 Community and Visual Studio Professional 2017, and on AppVeyor.

File: src/Imgur.API/Endpoints/Impl/EndpointBase.cs [hunk 1]
```diff
@@ -30,7 +30,9 @@ public abstract class EndpointBase : IEndpoint
         protected EndpointBase(IApiClient apiClient)
             : this(apiClient, new HttpClient())
         {
-            ApiClient = apiClient ?? throw new ArgumentNullException(nameof(apiClient));
+            if(apiClient == null)
```
Comment (DamienDennehy) [inline src/Imgur.API/Endpoints/Impl/EndpointBase.cs:33]: This is an older C# standard, not a newer one.

File: src/Imgur.API/Endpoints/Impl/EndpointBase.cs [hunk 2]
```diff
@@ -69,10 +71,10 @@ internal EndpointBase(IApiClient apiClient, HttpClient httpClient)
                     : $"Client-ID {apiClient.ClientId}");
 
             //Add Mashape Authentication header
-            if (apiClient is IMashapeClient mashapeClient)
+            if (apiClient is IMashapeClient)
```
Comment (DamienDennehy) [inline src/Imgur.API/Endpoints/Impl/EndpointBase.cs:74]: My code is a C# 7 feature called Pattern Matching. I'm not sure why you're trying to revert it.
https://csharp.christiannagel.com/2017/02/15/patternmatching/

**Review by DamienDennehy — COMMENTED** [review]
The project file is a .NET Standard Library and is not out of date.

image

**Comment by master117** [comment]
My bad, http://structuredsight.com/2016/09/01/c-7-additions-throw-expressions/ you are correct ?? now works since throw became an expression. Same for out https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out-parameter-modifier and is. Apparently I used 2015 to check twice instead of 2017. This code won't run on 2015 and C#6.0 an earlier tho, just as a heads up.

**Pull request closed without merging**